src/core_types.cpp

```diff
@@ -95,6 +101,13 @@ Capabilities::proposals_supported(
     });
 }
 
+bool
+Capabilities::credential_supported(const Credential& credential) const
+{
+  const auto cred_type = credential.type();
+  return std::count(credentials.begin(), credentials.end(), cred_type) > 0;
+}
```

Comment:
Nit .. may be simplify it to one line ?

Reply:
`clang-format` was wrapping it, so i thought it would be a little more elegant this way.